### ISSUE TITLE
Remote sync not working??

### ISSUE THREAD

**Issue by rickster04** [yongkangchen/remote-sync#204]
Hi, this looks like a great utility with a lot of downloads, but it doesn't seem to work and I have no idea why.  there are no error messages or other notifications.  Any help greatly appreciated!  thanks

**Comment by chris-aeviator** [comment]
It took me a little while to get it working as well so i'll shortly describe the workflow as the documentation is lacking a tutorial.

First step you need to do after you installed the plugin, is to go to your local Development Folder (e.g. ``/var/www/``) and create a new folder with the **exact** name of the remote folder you want to have in sync. 

Let's say you want to remote-sync a wordpress folder that's running in a vagrant machine ( the remote folder in this example is ```/home/vagrant/webapps/wp``). 

go into your newly created (empty) folder on your local machine ```cd /var/www/wp```.
Now you need to create the config file ```.remote-sync.json``` and fill in the right settings, for the vagrant example it would be something like this:

```json
{
  "uploadOnSave": true,
  "hostname": "192.168.33.10",
  "target": "/home/vagrant/webapps/wp",
  "ignore": [
    ".git/**"
  ],
  "username": "vagrant",
  "password": "vagrant",
  "transport": "scp"
}
```
Go ahead and save the file.
From this point the remote-sync plugin tries to upload the json file to the remote machine ( you can check in the small console that should appear in the downer part of atom or check directly in your remote machine if the json file appears in your wp directory ).

Now you can try to download the remote folder by right-clicking the wp folder in atom -> remote sync -> download folder.

Once the sync has succeeded the first time everything should run automatic.

Hope this helps, it's a great great tool!

**Comment by rickster04** [comment]
Many thanks for the response Chris.

I have made a little progress.

I changed the target directory in the .remote-sync.json file to be the same in both local and remote machines.  I could not get Remote-Sync to copy the .remote-sync.json file over to the remote server (even though it had done this before) so I copied it over manually via FTP.

I then opened the local folder in Atom, right clicked on it and selected Remote-Sync -> Download Folder.

The panel at the bottom of the Atom screen opened, and it said

Downloading all files: /xxx/xx/xxx/xx
Connecting xxx@xx.xx.xx.xx:21
(which looks fine)

then I get a lot of error messages:

Error: Cannot open data connection.
Error: Unable to make data connection
Error: Unable to make data connection
Error: Unable to make data connection
Error: Cannot open data connection.
Error: Unable to make data connection
Error: Unable to make data connection
Error: Cannot open data connection.
Error: Unable to make data connection
Error: Cannot open data connection.
Error: Cannot open data connection.
Error: Cannot open data connection.
Error: Cannot open data connection.
Error: Unable to make data connection

I have double checked the FTP credentials in the config file and they are correct.

Any further help gratefully received!

thanks

R

**Comment by yongkangchen** [comment]
I guess there are some problems with your ftp server.
Please try to google: `ftp Cannot open data connection`

**Comment by rickster04** [comment]
Thanks YongKang.  I see someone else has the same problem //91.

I tried googling the error message but all responses relate to other apps not Atom so I'm none the wiser.

I'm pretty sure my FTP server is working fine.  I never have any problems with it when using other FTP clients.

Thanks anyway - would love to get this working but I understand if you're too busy to deal with this problem.

**Comment by DISKONEKTeD** [comment]
Actually, i got it to work just now, on the right hand corner, there was a red error (of atom) and so i clicked it. It opened a new tab: It said something about a file being from a previous build of atom and thus it was not compatible. So I chose to rebuild option (at the top right corner), and it did so successfully. Then I clicked on the reload atom option (again, top right corner). Once it reloaded I then proceed with the right click config on the project folder and the missing menu finally appeared.
